src/Cache/ShardedPhpRedis.php

```diff
@@ -0,0 +1,326 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\redis\Cache\PhpRedis.
+ */
+
+namespace Drupal\redis\Cache;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Cache\CacheTagsChecksumInterface;
+
+/**
+ * PhpRedis cache backend.
+ */
+class ShardedPhpRedis extends CacheBase {
```

Comment:
From what I read you are still using some advanced Redis features that would not pass with a sharding proxy, but that will be really a lot faster when using a single server, so your implementation may actually be useful for people using a single instance.